microprofile/WFLY-14987_reactive_messaging-ssl-context-kafka-connector.adoc

```diff
@@ -0,0 +1,272 @@
+** If there is no Elytron provided `SSLContext` with that name an error will be thrown on deployment
+* The examples shown above for using properties from a deployment `microprofile-config.properties`. However, the configuration is using MicroProfile Config, so other mechanisms to specify these values, such as environment variables, are also valid.
+* The indicated Elytron `SSLContext` must be present at the time of deployment, or an error will be thrown.
+* There will be a dependency from the deployment on the JBoss MSC service providing the `SSLContext`.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+* Although we encourage users to use the Elytron provided `SSLContext` mechanism, it is not an error to use the native SmallRye Reactive Messaging/Apache Kafka client properties
+* If the user has specified the `<PREFIX>wildfly.elytron.ssl.context` property in their MicroProfile Config, it is not an error to use the other Kafka native ssl properties (e.g. the ones that start with `<PREFIX>.ssl.` in the configuration). These properties will simply be ignored.
+
+== Test Plan
+At the moment, it looks like the tests for this will need to reside in QE's testsuite.
+
+The issue is that we are currently using the Spring embedded Kafka test utility for testing the Kafka integration which does not have the required hooks. testcontainers.org is also popular but requires users have a running version of Docker on their machines, which is not the case for our CI.
+
+I looked into forking the Spring utility, but noticed that the system property needed to set up JAAS (`-Djava.security.auth.login.config`) is read at JVM boot. Also, Zookeeper and Kafka need a separate config file for JAAS, and that config file must be specified on JVM startup via a system property. Hence enhancing the embedded Kafka used by the current WildFly tests does not look practical at the moment without writing a lot of process wrappers and starting them as external processes.
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
+////
+The WildFly documentation will be updated to cover this use case.
+
+== Example Setup and Configuration
+For manual testing, I use Zookeeper and Kafka from the download as indicated in the https://kafka.apache.org/quickstart[Apache Kafka Quickstart].
+
+=== Generating the key-/truststores
+To create the certificates:
+Followed  https://docs.actian.com/integrationmanager/2.0/index.html#page/User/Step_1_3a_Create_a_Keystore_File.htm
+
+Create the server keystore with a certificate
+keytool -genkeypair -alias localhost -keyalg RSA -keysize 2048 -storetype PKCS12 -keystore server.keystore.p12 -validity 3650  -ext SAN=DNS:localhost,IP:127.0.0.1
+(Pwd: serverks)
```

Comment:
would be super nice to put this command (and also below commands) into the `command block`

Reply:
Done, and I think I've addressed the rest